openpype/hosts/maya/api/lib.py

```diff
@@ -3053,31 +3061,52 @@ def remove_render_layer_observer():
         pass
 
 
-def update_content_on_context_change():
-    """
-    This will update scene content to match new asset on context change
+def update_instances_frame_range():
+    """Update 'frameStart', 'frameEnd', 'handleStart', 'handleEnd' and 'asset'
+    attributes of sets that got one, execpt if instance family is 'render'
     """
-    scene_sets = cmds.listSets(allSets=True)
+    collected_instances = cmds.ls(
+        "*.id",
+        long=True,
+        type="objectSet",
+        recursive=True,
+        objectsOnly=True
+    )
     asset_doc = get_current_project_asset()
     new_asset = asset_doc["name"]
     new_data = asset_doc["data"]
-    for s in scene_sets:
-        try:
-            if cmds.getAttr("{}.id".format(s)) == "pyblish.avalon.instance":
-                attr = cmds.listAttr(s)
-                print(s)
-                if "asset" in attr:
-                    print("  - setting asset to: [ {} ]".format(new_asset))
-                    cmds.setAttr("{}.asset".format(s),
-                                 new_asset, type="string")
-                if "frameStart" in attr:
-                    cmds.setAttr("{}.frameStart".format(s),
-                                 new_data["frameStart"])
-                if "frameEnd" in attr:
-                    cmds.setAttr("{}.frameEnd".format(s),
-                                 new_data["frameEnd"],)
-        except ValueError:
-            pass
+
+    frames_attributes = {
+        'frameStart': new_data["frameStart"],
+        'frameEnd': new_data["frameEnd"],
+        'handleStart': new_data["handleStart"],
+        'handleEnd': new_data["handleEnd"],
+        'asset': new_asset
+    }
+
+    for instance in collected_instances:
+        family_attr = "{}.family".format(instance)
+        if (cmds.attributeQuery('family', node=instance, exists=True) and
+                cmds.getAttr(family_attr) == "render"):
+            continue
+
+        id_attr = "{}.id".format(instance)
+        if cmds.getAttr(id_attr) != "pyblish.avalon.instance":
+            continue
```

Comment:
```suggestion
        id_attr = "{}.id".format(instance)
        if cmds.getAttr(id_attr) != "pyblish.avalon.instance":
            continue
            
        if (
            cmds.attributeQuery('family', node=instance, exists=True) and
            cmds.getAttr("{}.family".format(instance)) == "render"
        ):
            continue
```

Re-ordered to check the `.id` attribute value first since if it's true it's also very likely it has the family attribute instead of first checking the family attribute on nodes with an `.id` attribute that might not be a publish instance.

I wonder if this way of writing is more readable for the family attribute if statement - so I've refactored that too. I believe this would also only format the string if the attribute existed to begin with.

Thoughts?